libs/common-services/src/test/java/uk/gov/di/ipv/core/library/helpers/VerifiableCredentialGenerator.java

```diff
@@ -132,7 +144,10 @@ private static String signTestJWT(JWTClaimsSet claimsSet) throws Exception {
         SignedJWT signedJWT = new SignedJWT(jwsHeader, claimsSet);
         signedJWT.sign(ecdsaSigner);
 
-        return signedJWT.serialize();
+        // Parsing and serialising here allows us to cast like before.
+        // The casting will be replaced in PYIC-5171 and so should this.
```

Comment:
This comes from casting like this:

```java
JSONObject vcClaim = (JSONObject) vc.getClaimsSet().getClaim(VC_CLAIM);
JSONArray evidenceArray = (JSONArray) vcClaim.get(VC_EVIDENCE);
```

For some reason the `signedJwt` in this method needs to be parsed for this casting to work. 

- Before, that would happen because this is used in generating a string which would have to be `SignedJWT.parse`ed later on, 
- but now I'm adding in this to ensure it gets parsed, and since it is destined to be replaced when we don't need to cast (PYIC-5171)

Reply:
Ah - it'll be because when you're building the `JWTClaimsSet` you're not using the JSON classes (we're using `TestVc`), but by serializing and reparsing, it parses it into the real JSON object.

To do it 'properly' you should make sure that all the claims you put onto the `JWTClaimsSet` are the appropriate JSON types, but doing this is a fairly easy hack to do them all at once :shrug: